### PR TITLE
Add explain command

### PR DESCRIPTION
**Pull request by kt-eliatra**
### Description
Add explain command

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/556

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by YANG-DB** [comment]
Please add command usage examples in the main readme markdown 
Thanks

**Comment by kt-eliatra** [comment]
> Please add command usage examples in the main readme markdown Thanks

done

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java [hunk 1]
```diff
@@ -123,7 +125,8 @@ public LogicalPlan visitQuery(Query node, CatalystPlanContext context) {
 
     @Override
     public LogicalPlan visitExplain(Explain node, CatalystPlanContext context) {
-        return node.getStatement().accept(this, context);
+        node.getStatement().accept(this, context);
+        return context.apply(p -> new ExplainCommand(p, ExplainMode.fromString("simple")));
```
Comment (YANG-DB) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:129]: what other explain types can we engage ?

**Review by YANG-DB — COMMENTED** [review]
Can you please review the [explain options](https://spark.apache.org/docs/latest/sql-ref-syntax-qry-explain.html) ?
and also add an explain test to validate the expected result ? - it would be nice to add these parameters for the user to request...

**Comment by kt-eliatra** [comment]
> Can you please review the [explain options](https://spark.apache.org/docs/latest/sql-ref-syntax-qry-explain.html) ? and also add an explain test to validate the expected result ? - it would be nice to add these parameters for the user to request...

the explain command supports now `explain mode` argument. I also wrote tests for each mode.

**Review by YANG-DB — APPROVED** [review]

**Pull request merged**